File: WebService/Cliente/Models/Reserva.cs
using System;
using System.Runtime.Serialization;

namespace RazorPagesMovie.Models
{
    [DataContract(Namespace="")]
    public class Reserva
    {
        [DataMember(Order=1)]
        public string nome { get; set; }
        [DataMember(Order=2)]
        public string destino { get; set; }
        [DataMember(Order=3)]
        public int nQuartos { get; set; }
        [DataMember(Order=4)]
        public int nPessoas { get; set; }

        public Reserva(string nome, string destino, int nQuartos, int nPessoas)
        {
            this.nome=nome;
            this.destino=destino;
            this.nQuartos=nQuartos;
            this.nPessoas=nPessoas;
        }
    }
}

File: WebService/WebServer/src/java/recursos/Reserva.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package recursos;

import java.io.Serializable;
import java.time.LocalDate;
import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;

/**
 *
 * @author a1717553
 */
@XmlRootElement(name="Reserva")
@XmlAccessorType(XmlAccessType.FIELD)
public class Reserva implements Serializable{
    @XmlElement(required=true)
    public String nome;
    @XmlElement(required=true)
    public String destino;
    @XmlElement(required=true)
    public int nQuartos;
    @XmlElement(required=true)
    public int nPessoas;
    
    public Reserva(){}
    public Reserva(String n, String d, int q, int p){
        this.nome=n;
        this.destino=d;
        this.nQuartos=q;
        this.nPessoas=p;
    }
}
